mypy/constant_fold.py

```diff
@@ -102,25 +148,41 @@ def constant_fold_binary_int_op(op: str, left: int, right: int) -> int | None:
     return None
 
 
-def constant_fold_unary_int_op(op: str, value: int) -> int | None:
-    if op == "-":
-        return -value
-    elif op == "~":
-        return ~value
-    elif op == "+":
-        return value
+def constant_fold_binary_float_op(op: str, left: int | float, right: int | float) -> float | None:
+    assert not (isinstance(left, int) and isinstance(right, int)), (op, left, right)
+    if op == "+":
+        return left + right
+    elif op == "-":
+        return left - right
+    elif op == "*":
+        return left * right
+    elif op == "/":
+        if right != 0:
+            return left / right
+    elif op == "//":
+        if right != 0:
+            return left // right
+    elif op == "%":
+        if right != 0:
+            return left % right
+    elif op == "**":
+        if (left < 0 and right >= 1 or right == 0) or (left >= 0 and right >= 0):
```

Comment:
This could return a complex number, but the return type doesn't include `complex` (not sure if supporting complex results is worth it):
```
>>> (-1.2) ** 1.5
(-2.414760036730213e-16-1.3145341380123985j)
```
I used this fragment to look for other interesting cases, maybe it's helpful here:
```py
values = -1.5, -1.0, -0.5, 0.0, 0.5, 1.0, 1.5
for x in values:
    for y in values:
        try:
            print(x, y, x**y)
        except Exception:
            print(f'error: {x} ** {y}')
```